src/compiler/program.ts

```diff
@@ -454,13 +454,26 @@ module ts {
         }
 
         function verifyCompilerOptions() {
-            if (!options.sourceMap && (options.mapRoot || options.sourceRoot)) {
+            if (!options.sourceMap && (options.mapRoot || options.sourceRoot || options.inlineSourceMap)) {

```

Comment:
I'm sure I was the one that wrote it, but we could just factor the `return` out into each check instead of checking each one twice. I don't really mind though.